### ISSUE TITLE
500 error - Unable to obtain lock for project in file for project

### ISSUE THREAD

**Issue by tetchel** [eclipse/codewind#1910]
Sometimes when connecting to a remote Codewind instance my requests fail with the following message:
```
StatusCodeError: 500 - {"name":"ProjectError","code":"LOCK_FAILURE","info":{"name":"ProjectError","code":"LOCK_FAILURE","message":"Unable to obtain lock for project in file for project nodde123123."}}
```

I have seen this when calling the `/logs` and `/metrics/status` endpoints for a given project, which are the two requests performed when the IDE becomes aware of a project (capabilities used to be too, but that one is now delayed since #1073). These requests are performed in parallel. So likely one of them is blocking the other while PFE [waits for the inf to unlock](https://github.com/eclipse/codewind/blob/master/src/pfe/portal/modules/Project.js#L241).

**Comment by rwalle61** [comment]
The new sync tests are exposing this problem too - they call `upload/end`. See https://ci.eclipse.org/codewind/job/Codewind/job/codewind/job/PR-1899/3/consoleFull. I'm investigating why that happens in that context

**Comment by rwalle61** [comment]
The locking mechanism is currently designed to try to get a lock for 1 second then throw this error. Perhaps we need to extend this time, or maybe it should never take this long and we have a bug causing it to take so long

**Comment by hhellyer** [comment]
It shouldn't take long to write to the file, I think if we extend the lock then we risk just creating a longer queue.

It might be worth reviewing whether this is still a problem after #1875 is fixed as we may be holding the lock for longer if we are trying to write more fields to the project.inf file. Resloving that may also resolve this.

**Comment by tetchel** [comment]
fwiw i only saw this on a remote connection, which makes sense because it would take longer to update the file when we're writing to NFS storage instead of a regular disk

**Comment by tetchel** [comment]
Still seeing it

**Comment by rwalle61** [comment]
I saw this on a Jenkins PR build today, while a PFE project API test (`project/{id}/metrics/inject`) was trying to remove an open liberty project (in the clean up phase of a test)

**Comment by tetchel** [comment]
I don't see this anymore 🤷‍♂

**Comment by rwalle61** [comment]
It's good you're not seeing this error in use anymore, however we're still seeing this in the PFE PR build tests, e.g. https://ci.eclipse.org/codewind/job/Codewind/job/codewind/job/master/639/execution/node/65/log/ 

Might be worth (1) reopening this issue, or (2) use this issue to track seeing the error in the IDE, and open a new issue to track seeing the error in the PFE PR build

**Comment by tetchel** [comment]
I think 2) is a good idea
